libnavigation-core/src/test/java/com/mapbox/navigation/core/trip/session/MapboxTripSessionTest.kt

```diff
@@ -764,6 +781,39 @@ class MapboxTripSessionTest {
             assertEquals(tripSession.primaryRoute, refreshedRoutes.first())
         }
 
+    @Test
+    fun `route set result - native alternatives successful for primary and failed for alternatives`() =
```

Comment:
I think yes. Every call to `Navigator#refreshRoute` updates state in NN. The last successful result should contain metadata of the most actual state from NN. To keep things consistent in case of partial error, I would update metadata with the latest result from NN.

Reply:
Updated.